snapr/src/builder.rs

```diff
@@ -1,8 +1,65 @@
 use std::fmt;
 
+use macros::impl_snapr_builder;
+
 use crate::{Error, Snapr, TileFetcher, Zoom};
 
-/// Builder structure for [`snapr`].
+pub(crate) mod macros {
+    macro_rules! impl_snapr_builder {
+        (($builder_ty: ty, $builder_ident: ident), ($snapr_ty: ty, $snapr_ident: ident), ($tile_fetcher_ty: ty, $tile_fetcher_ident: ident)) => {
```

Comment:
I wonder if there's a macro that strips out lifetimes from `ty` tokens when calling the `stringify!` macro... that'd be really nice here.